src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -428,8 +439,7 @@ def __init__(self, auth_ctx_factory=None):
                             os.path.join(get_config_dir(), 'accessTokens.json'))
         self._service_principal_creds = []
         self._auth_ctx_factory = auth_ctx_factory or _AUTH_CTX_FACTORY
-        self.adal_token_cache = None
-        self._load_creds()
+        self._adal_token_cache_attr = None
```

Comment:
I don't think we need `_attr` suffix and I don't see we have other code using this convention.  PEP 8 calls out we only need a leading `_`. But I am not going to have a naming convention discussion here :), if you feel it fine, i am fine with it too.